handlers/soft-opt-in-consent-setter/src/main/scala/com/gu/soft_opt_in_consent_setter/SfQueries.scala

```diff
@@ -36,6 +36,31 @@ object SfQueries {
     """.stripMargin
   }
 
+  val getProductSwitchSubsToProcessQuery: String = {
+    val limit = 50
+
+    s"""
+       |SELECT
+       |    Id,
+       |    Name,
+       |    Product__c,
+       |    SF_Subscription__r.SF_Status__c,
+       |    SF_Subscription__r.Soft_Opt_in_Status__c,
+       |    SF_Subscription__r.Soft_Opt_in_Last_Stage_Processed__c,
+       |    SF_Subscription__r.Soft_Opt_in_Number_of_Attempts__c,
+       |    Buyer__r.IdentityID__c
+       |FROM
+       |    Subscription_Rate_Plan_Update__c
+       |WHERE
+       |    SF_Subscription__r.Soft_Opt_in_Number_of_Attempts__c < 5 AND
+       |    SF_Subscription__r.Soft_Opt_in_Eligible__c = true AND
+       |    SF_Subscription__r.SF_Status__c = 'Active' AND
+       |    SF_Subscription__r.Soft_Opt_in_Last_Stage_Processed__c = 'Acquisition'
+       |LIMIT
```

Comment:
This logic would mean that a subscription could only be processed as a "switch" once. If we introduce further switch journeys (e.g. Supporter Plus to Recurring Contribution) it may be necessary to facilitate multiple "switch" soft opt-in processing.

This might mean having an additional `Soft_Opt_in_Processed__c` field on `Subscription_Rate_Plan_Update__c` which is updated in Salesforce record writeback so that we know whether a specific switch has been processed.

On the other hand, this might be ok as a V1 as it wouldn't likely be an issue until more switch journeys are enabled.

Reply:
True, let's do that as part of this PR, do you mind adding the field?

Reply:
Sure, I think we should also add a `Soft_Opt_in_Number_of_Attempts__c` to `Subscription_Rate_Plan_Update__c` then on a failure you can write back to this in a similar way. What do you think?

I'll create a `Soft_Opt_in_Processed__c` field as a boolean, so on successful writeback you should change this to `TRUE`.

This would change the query to:

```suggestion
       |WHERE
       |    Soft_Opt_in_Number_of_Attempts__c < 5 AND
       |    Soft_Opt_in_Processed__c = false
       |    SF_Subscription__r.SF_Status__c = 'Active' AND
       |LIMIT
```

Reply:
sounds good

Reply:
You should now see `Soft_Opt_In_Processed__c` (boolean) and `Soft_Opt_in_Number_of_Attempts__c` (integer) in the DEV Salesforce sandbox